File: enterprise/backend/src/metabase_enterprise/enhancements/integrations/google.clj
(ns metabase-enterprise.enhancements.integrations.google
  (:require
   [metabase.settings.core :as setting :refer [define-multi-setting-impl]]))

(define-multi-setting-impl google-auth-auto-create-accounts-domain :ee
  :getter (fn [] (setting/get-value-of-type :string :google-auth-auto-create-accounts-domain))
  :setter (fn [domain] (setting/set-value-of-type! :string :google-auth-auto-create-accounts-domain domain)))

File: frontend/src/metabase/api/google.ts
import type { EnterpriseSettings } from "metabase-types/api";

import { Api } from "./api";
import { invalidateTags, tag } from "./tags";

type GoogleAuthSettings = Pick<
  EnterpriseSettings,
  | "google-auth-enabled"
  | "google-auth-client-id"
  | "google-auth-auto-create-accounts-domain"
>;

export const googleApi = Api.injectEndpoints({
  endpoints: (builder) => ({
    updateGoogleAuth: builder.mutation<void, GoogleAuthSettings>({
      query: (settings) => ({
        method: "PUT",
        url: `/api/google/settings`,
        body: settings,
      }),
      invalidatesTags: (_, error) =>
        invalidateTags(error, [tag("session-properties")]),
    }),
  }),
});

export const { useUpdateGoogleAuthMutation } = googleApi;
